App/Redux/ProcessingImagesRedux.ts

```diff
@@ -0,0 +1,207 @@
+import { createAction, ActionType, getType } from 'typesafe-actions'
+import { SharedImage, AddResult } from '../Models/TextileTypes'
+import { RootState } from './Types'
+
+const actions = {
+  insertAddingImage: createAction('processingImages/INSERT_ADDING_IMAGE', resolve => {
+    return (sharedImage: SharedImage, destinationThreadId: string, comment?: string) => resolve({ sharedImage, destinationThreadId, comment })
+  }),
+  imageAdded: createAction('processingImages/IMAGE_ADDED', resolve => {
+    return (sharedImage: SharedImage, addResult: AddResult) => resolve({ sharedImage, addResult })
+  }),
+  addingError: createAction('processingImages/ADDING_ERROR', resolve => {
+    return (sharedImage: SharedImage, error: any) => resolve({ sharedImage, error })
+  }),
+  imageUploadProgress: createAction('processingImages/IMAGE_UPLOAD_PROGRESS', resolve => {
+    return (dataId: string, progress: number) => resolve({ dataId, progress })
+  }),
+  imageUploadComplete: createAction('processingImages/IMAGE_UPLOAD_COMPLETE', resolve => {
+    return (dataId: string, responseCode: string, responseBody: string) => resolve({ dataId, responseCode, responseBody })
+  }),
+  imageUploadError: createAction('processingImages/IMAGE_UPLOAD_ERROR', resolve => {
+    return (dataId: string, errorMessage: string) => resolve({ dataId, errorMessage })
+  }),
+  addingToWallet: createAction('processingImages/ADDING_TO_WALLET', resolve => {
+    return (dataId: string) => resolve({ dataId })
+  }),
+  sharingToThread: createAction('processingImages/SHARING_TO_THREAD', resolve => {
+    return (dataId: string) => resolve({ dataId })
+  }),
+  sharingError: createAction('processingImages/SHARING_ERROR', resolve => {
+    return (dataId: string, error: any) => resolve({ dataId, error })
+  }),
+  complete: createAction('processingImages/COMPLETE', resolve => {
+    return (dataId: string) => resolve({ dataId })
+  })
+}
+
+export type ProcessingImagesAction = ActionType<typeof actions>
+
+export type ProcessingImage = {
+  readonly sharedImage: SharedImage
+  readonly destinationThreadId: string
+  readonly comment?: string
+  readonly state: 'adding' | 'uploading' | 'addingToWallet' | 'sharing'
+  readonly error?: string
+  readonly addData?: {
+    readonly addResult: AddResult
+  }
+  readonly uploadData?: {
+    readonly uploadProgress: number
+    readonly responseCode?: string
+    readonly responseBody?: string
+    readonly errorMessage?: string
+  }
+}
+
+export type ProcessingImagesState = {
+  readonly images: ProcessingImage[]
+}
```

Comment:
This redux is where we track and persist the state of images from when they are first picked to be shared until they are done being shared to a thread. We track state though all of adding to IPFS, upload for remote pinning, adding to wallet (`default` thread), and sharing to thread.

Reply:
nice